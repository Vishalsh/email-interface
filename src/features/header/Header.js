import React from 'react';
import PropTypes from 'prop-types';

import Button from 'components/button/Button';
import classes from './Header.module.scss';

const Header = ({ onClickHamburger }) => {
  return (
    <header className={classes.header}>
      <div className="row">
        <div className="col-xs-6">
          <div className="row middle-xs">
            <Button type="primary" onClick={onClickHamburger}>
              <i className="icon icon-hamburger"/>
            </Button>
            <input className={classes.search} placeholder="Search for something..."/>
          </div>
        </div>

        <div className="col-xs-6">
          <div className="row end-xs middle-xs">
            <i className="icon icon-email-solid"/>
            <div>
              <Button onClick={() => alert('logout')}>
                <i className="icon icon-logout"/>
              </Button>
              <span>Log out</span>
            </div>
          </div>
        </div>
      </div>
    </header>
  );
};

Header.propTypes = {
  onClickHamburger: PropTypes.func.isRequired,
};

export default Header;
