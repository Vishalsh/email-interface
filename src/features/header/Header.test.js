import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Header from './Header';

describe('Header', () => {
  it('should render the component', () => {
    expect(toJson(shallow(<Header/>))).toMatchSnapshot();
  })
});