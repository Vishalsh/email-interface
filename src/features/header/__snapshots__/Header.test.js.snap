// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render the component 1`] = `
<header
  className="header"
>
  <div
    className="row"
  >
    <div
      className="col-xs-6"
    >
      <div
        className="row middle-xs"
      >
        <Button
          disabled={false}
          onClick={[Function]}
          type="primary"
        >
          <i
            className="icon icon-hamburger"
          />
        </Button>
        <input
          className="search"
          placeholder="Search for something..."
        />
      </div>
    </div>
    <div
      className="col-xs-6"
    >
      <div
        className="row end-xs middle-xs"
      >
        <i
          className="icon icon-email-solid"
        />
        <div>
          <Button
            disabled={false}
            onClick={[Function]}
            type="default"
          >
            <i
              className="icon icon-logout"
            />
          </Button>
          <span>
            Log out
          </span>
        </div>
      </div>
    </div>
  </div>
</header>
`;
